File: business/business-oj/src/main/java/com/glowxq/oj/problem/enums/ProgramType.java
package com.glowxq.oj.problem.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.core.common.exception.common.AlertsException;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/3/18
 */
@Getter
@AllArgsConstructor
public enum ProgramType implements BaseType {
    ACM(0, "acm"),
    OI(1, "oi");

    private final Integer type;

    private final String name;

    /**
     * 根据code获取枚举
     *
     * @return 对应的枚举实例，未找到返回null
     */
    public static ProgramType matchCode(Integer code) {
        for (ProgramType type : ProgramType.values()) {
            if (type.getCode().equals(code)) {
                return type;
            }
        }
        throw new AlertsException("编程题类型转换错误:" + code);
    }

    @Override
    public Object getCode() {
        return type;
    }
}

File: oj-web/src/enums/oj/problem/ProgramType.ts
import type { BaseEnum } from '@/enums/base';

// 程序类型枚举接口
interface ProgramTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 程序类型枚举
 * 用于标识不同的程序类型
 */
export const ProgramType: { [key: string]: ProgramTypeEnum } = {

  ACM: {
    code: 0,
    name: 'ACM',
    tooltip: 'ACM',
    text: 'ACM程序'
  },

  OI: {
    code: 1,
    name: 'OI',
    tooltip: 'OI',
    text: 'OI程序'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: number): ProgramTypeEnum | null {
  return Object.values(ProgramType).find(item => item.code === code) || null;
}
